### PR TITLE
fix(common): don't convert null to a string when flushing a mock request

### LINKED ISSUE #20744: [Regression] HttpClient converts null -> 'null' in unit tests: "Expected 'null' to be null."
## I'm submitting a...

<pre><code>
[ X ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

In version 5.0.4 and 5.0.5, during unit tests, flushing http requests with a `null` value converts the result into `'null'` (string)

In the test this causes the error: `ApiService should return null FAILED  Expected 'null' to be null.`

## Expected behavior

In version 5.0.3 flushing http requests from http requests with a `null` value returned a `null` value. This is a regression.

## Minimal reproduction of the problem with instructions

I made a repo but you can also find the relevant code below.

To reproduce in an [example repo](https://github.com/danwulff/http-client-test). In a terminal run:
`git clone https://github.com/danwulff/http-client-test.git`
`cd http-client-test && npm i`
`ng test`

Spec [src/app/service/api.service.spec.ts](https://github.com/danwulff/http-client-test/blob/master/src/app/service/api.service.spec.ts):
```
import { TestBed, inject } from '@angular/core/testing';
import { HttpClientTestingModule, HttpTestingController, TestRequest } from '@angular/common/http/testing';

import { ApiService } from './api.service';

fdescribe('ApiService', () => {

  let httpMock: HttpTestingController;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientTestingModule],
      providers: [ApiService]
    });
    httpMock = TestBed.get(HttpTestingController);
  });

  it('should be created', inject([ApiService], (service: ApiService) => {
    expect(service).toBeTruthy();
  }));

  it('should return null', inject([ApiService], (service: ApiService) => {
    const something = { yup: 'huh' };

    let actualResponse: void;

    service.postSomething(something).subscribe(response => {
      actualResponse = response;
    });
    const req: TestRequest = httpMock.expectOne(`/rest/post-a-thing`);
    expect(req.request.method).toBe('POST');
    expect(req.request.body).toBe(something);
    req.flush(null);

    expect(actualResponse).toBeNull();  // <- this causes the error: Expected 'null' to be null.
  }));
});
```

Service [src/app/service/api.service.ts](https://github.com/danwulff/http-client-test/blob/master/src/app/service/api.service.ts):
```
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs/Observable';

@Injectable()
export class ApiService {

  constructor(public http: HttpClient) { }

  postSomething(thing: object): Observable<void> {
    return this.http.post<void>(`/rest/post-a-thing`, thing);
  }
}
```

## What is the motivation / use case for changing the behavior?

Backwards compatibility

## Environment

<pre><code>
Angular version: 5.0.4 & 5.05

Browser:
- [ X ] Chrome (desktop) version 62
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 6.12.0  
- Platform:  Linux & Windows 

Others:

</code></pre>

### PR DESCRIPTION
**Pull request by alxhub**
A bug in TestRequest caused null response bodies to be stringified. This
change causes null to be treated faithfully.

Fixes #20744

### CODE REVIEW

File: packages/common/http/testing/test/request_spec.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ddescribe, describe, iit, it} from '@angular/core/testing/src/testing_internal';
```
Comment (alfaproject) [inline packages/common/http/testing/test/request_spec.ts]: `iit`

File: packages/common/http/testing/BUILD.bazel [hunk 2]
```diff
@@ -5,7 +5,12 @@ load("//tools:defaults.bzl", "ts_library")
 ts_library(
     name = "testing",
     testonly = 1,
-    srcs = glob(["**/*.ts"]),
+    srcs = glob(
+        [
+            "*.ts",
```
Comment (mhevery) [inline packages/common/http/testing/BUILD.bazel]: Why was this change needed?

File: packages/common/http/testing/test/request_spec.ts (hunk 1 above)
Comment (mhevery) [reply packages/common/http/testing/test/request_spec.ts]: why do you need to import custom `describe` Why is the default one not sufficient?

File: packages/common/http/testing/BUILD.bazel (hunk 2 above)
Comment (alxhub) [reply packages/common/http/testing/BUILD.bazel]: This directory previously did not have tests, so this package globbed all the files together. Now that it has tests, we need to somehow exclude them from the build.

File: packages/common/http/testing/test/request_spec.ts (hunk 1 above)
Comment (alxhub) [reply packages/common/http/testing/test/request_spec.ts]: It's probably not necessary for this file, but the `common/http` tests all use the `testing_internal` functions, which do things like handle `Promise` returns, etc.

I could migrate them all off, but I'm not sure if any code changes would be necessary.

Comment (mhevery) [reply packages/common/http/testing/test/request_spec.ts]: We would like to get rid of these long term, so let's not add more. Could you remove them from this test.

**Review by mhevery — APPROVED** [review]
Please remove the `import {ddescribe, describe, iit, it} from '@angular/core/testing/src/testing_internal';`

LGTM otherwise.
